### PR TITLE
parse every businesstemplates translations when generationg view ref …

### LINKED ISSUE #827: [BusinessTemplates] VBP translations
If i create a businessEntity that is displayed by a businessTemplates that has multiple locale translations, only the VBP view reference for the BT defaultLocale is generated not other locales

### PR DESCRIPTION
**Pull request by Charlie-Lucas**
…with doctrine events

## Type
Bugfix 

## Purpose

Fixes #827 

## BC Break
NO

### CODE REVIEW

File: Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php [hunk 1]
```diff
@@ -112,40 +112,44 @@ public function updateBusinessPages($entity, BusinessEntity $businessEntity, Ent
...
-                if ($businessPage && !$scheduledForRemove) {
-                    $oldSlug = $businessPage->getSlug();
-                    $newSlug = $entity->getSlug();
+                    if ($businessPage && !$scheduledForRemove) {
+                        $oldSlug = $businessPage->getSlug();
+                        $newSlug = $entity->getSlug();
 
-                    $businessPage->setName($virtualBusinessPage->getName());
-                    $businessPage->setSlug($virtualBusinessPage->getSlug());
+                        $businessPage->setName($virtualBusinessPage->getName());
```
Comment (paulandrieux) [inline Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php]: Are you sure the name and slug of the business page will not be override by the ones of the last locale of the BusinessTemplate ? You never setup  the currentLocale of $businessPage so the second translation of the BusinessTemplate risk to erase the default locale for $businessPage

Comment (paulandrieux) [inline Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php]: Are you sure the name and slug of the business page will not be override by the ones of the last locale of the BusinessTemplate ? You never setup  the currentLocale of $businessPage so the second translation of the BusinessTemplate risk to erase the default locale for $businessPage

File: Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php [hunk 2]
```diff
@@ -228,6 +243,12 @@ public function postFlush(PostFlushEventArgs $eventArgs)
     private function updateViewReference(LifecycleEventArgs $eventArgs)
     {
         $entity = $eventArgs->getEntity();
+
+        //if entity is a translation, get its translatable entity
```
Comment (gregumo) [inline Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php]: @paulandrieux What do you think about that ?

**Comment by gregumo** [comment]
A new PR #852 has been merged in 2.2 with these changes

**Pull request closed without merging**